staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -129,7 +129,7 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (*authenticator.R
 	clientCertificateExpirationHistogram.Observe(remaining.Seconds())
 	chains, err := req.TLS.PeerCertificates[0].Verify(optsCopy)
 	if err != nil {
-		return nil, false, err
+		return nil, false, fmt.Errorf("verifying cert for %s/%s [%s/%s] failed: %w", req.TLS.PeerCertificates[0].Subject.CommonName, req.TLS.PeerCertificates[0].Subject.Organization, req.TLS.PeerCertificates[0].Issuer.CommonName, req.TLS.PeerCertificates[0].Issuer.Organization, err)
```

Comment:
SerialNumber (if set) and NotBefore time seems more useful than the organization for identifying the particular certificate

Reply:
does this dump PII into the logs? cc @mikedanese

Reply:
I wanted to dump the `CommonName` because that's what identifies the client, `Organization` was included to uniquely identify the `CommonName` (like a namespace). If you are tracing down a client, such information is useful to identify a particular microservice. From there you can check the certificate manually and verify notBefore or any other property. I don't think `SerialNumber` would allow me to resolve that microservice, although it is the unique ID. I am open to adding more information to the error message though.

> does this dump PII into the logs?

Is `CommonName` a concern?

Reply:
I know we echo username back to the user in some scenarios, and add it to audit logs, but I thought we had generally gotten it out of the apiserver logs, for PII reasons...

Reply:
I can go with the SerialNumber to get at least something to identify the failure although tracing it back implies having all client certificates, so you can't easily say who is having issues without knowing the full set.

Looks like we log CN just a few lines down in 
https://github.com/kubernetes/kubernetes/blob/ac4e7f51a863435a6d8b4a42702e2fb6ec588000/staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go#L210

Reply:
> Looks like we log CN just a few lines down in

that is for the x509 verifier, not the authenticator, and indicates a misconfiguration

Reply:
> tracing it back implies having all client certificates, so you can't easily say who is having issues without knowing the full set

that's true... looks like many issuers generate a random serial